include/cuco/bloom_filter_policy.cuh

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuco/detail/bloom_filter/bloom_filter_policy_impl.cuh>
+#include <cuco/hash_functions.cuh>
+
+#include <cuda/std/array>
+
+#include <cstdint>
+
+namespace cuco {
+
+// TODO Policies are meant as customization points so we need to ensure a user-defined policy
+// fulfills a certain concept.
```

Comment:
If anyone has a better idea how to implement a C++20-like concept with C++17 I would be very open to it.

Reply:
Pretty sure concepts can be partially imitated with SFINAE. I'd recommend against it, though, for maintainability reasons.